src/main/ruby/core/io.rb

```diff
@@ -158,7 +158,7 @@ def fill(io)
 
       begin
         buffer = Truffle::POSIX.read_string(io.descriptor, count)
-      rescue Errno::EAGAIN
+      rescue Errno::EAGAIN, IO::EAGAINWaitReadable
```

Comment:
This should just be `IO::WaitReadable` which is the common ancestor of all these kind of "try again" errnos.